torchbenchmark/util/kernels/triton_fused_attention.py

```diff
@@ -730,27 +730,31 @@ def forward(ctx, q, k, v, causal, sm_scale):
         desc_v = torch.tensor(desc_v, device=v.device)
         grid = lambda args: (triton.cdiv(q.shape[2], args["BLOCK_M"]), q.shape[0] * q.shape[1], 1)
         '''
-        desc_k = torch.empty((TMA_SIZE), device="cuda", dtype=torch.int8)
-        desc_v = torch.empty((TMA_SIZE), device="cuda", dtype=torch.int8)
-        desc_q = torch.empty((TMA_SIZE), device="cuda", dtype=torch.int8)
-        desc_o = torch.empty((TMA_SIZE), device="cuda", dtype=torch.int8)
+        #desc_k = torch.empty((TMA_SIZE), device="cuda", dtype=torch.int8)
+        #desc_v = torch.empty((TMA_SIZE), device="cuda", dtype=torch.int8)
+        #desc_q = torch.empty((TMA_SIZE), device="cuda", dtype=torch.int8)
+        #desc_o = torch.empty((TMA_SIZE), device="cuda", dtype=torch.int8)
+        q_buf = torch.empty(TMA_SIZE, device="cpu")
+        k_buf = torch.empty(TMA_SIZE, device="cpu")
+        o_buf = torch.empty(TMA_SIZE, device="cpu")
+        v_buf = torch.empty(TMA_SIZE, device="cpu")
```

Comment:
@embg Should we start using create_2d_tma_descriptor that returns a TmaDescKernelParam? In order to work with autotuning, we need to define the descriptors first, then update them in grid_tma. But it seems TmaDescKernelParam is not exposed, can we declare the descriptors here?

Reply:
One idea: create a dictionary outside the grid function, then update the dictionary inside the grid function?

```
descs = dict()
def grid(META):
    descs["k"] = create...
    descs["v"] = create...
    descs["q"] = create...
```

Reply:
BTW, I just landed grid_constant TMA upstream: https://github.com/triton-lang/triton/commit/c25f684bb08162f3b31d16a41f96e996b84851dd

So you can now work off master instead of my branch.